src/api/localStorage.js

```diff
@@ -0,0 +1,96 @@
+export class LocalStorage {
+  constructor() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  createUser(user) {
+    this.updateUsers();
+    user.id = this.generateId();
+    this.users.push(user);
+    this.setLocalStorageData();
+  }
+
+  setLocalStorageData() {
+    localStorage.setItem('UserAccounts', JSON.stringify(this.users));
+  }
+
+  updateUsers() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  getUser(id) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    return user;
+  }
+
+  getAllUsers() {
+    return this.users;
+  }
+
+  editUser(id, editName, editValue) {
+    try{
+      const editObj = {};
+      editObj[`${editName}`] = `${editValue}`;
+      const updatedUser = Object.assign(this.getUser(id), editObj);
+      this.deleteUser(id);
+      this.createUser(updatedUser);
+    } catch (err) {console.log('Error Editing User: ', err);}
+  }
+
+  replaceUser(id, update) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    const replacedUser = Object.assign(user, update);
+    return replacedUser;
+  }
+
+  deleteUser(id) {
+    this.updateUsers();
+    this.users.forEach((user, index) => {
```

Comment:
The loop is odd to me, though I don't know I'd say it's any less efficient than what I'd do. I'll give it to you anyway and you can see what you think.

```suggestion
    const userIndex = this.users.indexOf(this.users.find(user => user.id === id));
    this.users.splice(userIndex, 1);
    this.setLocalStorageData();
```